### ISSUE TITLE
Create factory for creating Feign clients manually exactly like they would have been created by the @FeignClient annotation

### ISSUE THREAD

**Issue by sdoeringNew** [spring-cloud/spring-cloud-openfeign#42]
# Enhancement

We generate our Feign client interfaces with [swagger-codegen](https://swagger.io/tools/swagger-codegen/). OpenAPI (formerly known as Swagger) is widely used to describe schemas for (RESTful) interfaces.  
So we have our defined OpenAPI schema and out of this schema the Feign client interface classes are generated.

Now we have those interfaces but we can't add the ```@FeignClient``` annotation because the classes are generated.  
Today we are creating the Feign clients manually like it's described [here](http://cloud.spring.io/spring-cloud-static/Camden.SR2/#_creating_feign_clients_manually).

But that leaves a bad vibe to us. We don't know for sure if we did it exactly like the ```FeignClientFactoryBean``` does. And we have to keep track on further development in later versions of Spring Cloud.

So a factory method would be nice that creates a Feign client exactly like it's been done by using the ```@FeignClient``` annotation.

This factory method shall get the target interface as input parameter as well as all the parameters from the annotation.

The factory will then be injected into a ```@Bean``` definition in a ```@Configuration``` class like this:

```Java
@Configuration
public class FeignConfig {
    @Bean
    public FeignClientInterface feignClient(final FeignClientFactory feignClientFactory) {
        return feignClientFactory.create(FeignClientInterface.class, "feign-client-name");
    }
```

Or perhaps as builder:

```Java
@Configuration
public class FeignConfig {
    @Bean
    public FeignClientInterface feignClient(final FeignClient.Builder feignClientBuilder) {
        return feignClientBuilder
                .targetClass(FeignClientInterface.class)
                .name("feign-client-name")
                .decode404(true)
                .primary(false)
                .build();
    }
}
```

**Comment by sdoeringNew** [comment]
I would be happy to create a patch.

Just point me into the right direction and show me a good starting point.

**Comment by ryanjbaxter** [comment]
Patches are welcome, take a look at our contribution guidelines to get started 
https://github.com/spring-cloud/spring-cloud-openfeign/blob/master/.github/CONTRIBUTING.md

**Comment by ryanjbaxter** [comment]
Closed via  750cb35

**Comment by spencergibb** [comment]
That wasn't part of this patch. Please ask questions on stack overflow or gitter, not closed issues
